daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SExpr.scala

```diff
@@ -456,6 +464,18 @@ object SExpr {
       methodName: MethodName,
   ) extends SDefinitionRef
 
+  /** CoImplementsMethodDefRef(templateId, ref=ifaceId, method) invokes the
+    * interface-provided implementation of the method for the given template.
+    * Note that the order is (template, interface, method) like in
+    * 'ImplementsMethodDefRef'. This means that 'ref' must be the second
+    * argument, since a co-implementation is associated to the interface.
+    */
+  final case class CoImplementsMethodDefRef(
+      templateId: TypeConName,
+      ref: DefinitionRef,
```

Comment:
Looking at those types I find `DefinitionRef` weird for both `Implements` and `Coimplements`. If we look at the docs of `DefinitionRef` they state:

>   /** Reference to a value defined in the specified module. */

That’s a sensible definition but not one that applies here. Neither the template id nor the interface id are a value, they are type constructors. Is there a good reason to use `DefinitionRef` here? If not, it might be clearer to use `TypeConName` for both and call the fields `templateId` and `interfaceId`.

Reply:
The main use of `ref: DefinitionRef` seems to be defining the `packageName` and `modName` of the `SDefinitionRef`, which is used in [Speedy `lookupVal`](https://github.com/digital-asset/daml/blob/main/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala#L544-L563). I think it makes sense this way (at least for `ImplementsMethodDefRef` and `CoImplementsMethodDefRef`), since the parent package of an implemented resp. co-implemented method should be the template resp. interface package.